### ISSUE TITLE
New server wide setting to disable saving new dashboards from the UI 

### ISSUE THREAD

**Issue by holmesb** [grafana/grafana#32375]
Hi,
It would be reasonable to expect `allowUiUpdates: false` to disallow UI updates to dashboards.  Instead saving is allowed to dashboards that are created via the webui.

We have a gitops pipeline that manages dashboard changes declaratively.  But there's nothing stopping users from creating new dashboards using the UI and pressing the Save button to save.  We would prefer users to see the declarative save box:
![image](https://user-images.githubusercontent.com/5072156/112679101-b08eff00-8e63-11eb-8e53-66e5ce12f76e.png)

which is what they correctly see when they try to save a declaratively provisioned dashboard.  We have also tried setting `editable: false`. No difference.  Likewise no dice if we change user rights from Editor to Viewer - they can't create new dashboards at all, and don't see the Save button in existing ones.

Please make `allowUiUpdates: false` disallow UI updates.  Many thanks

**Comment by torkelo** [comment]
But it sounds like you want a new server level setting do disallow new dashboards from being saved from the UI?

**Comment by holmesb** [comment]
@torkelo - yes absolutely.  Creating new dashboards is fine, just saving them should be disallowed.  Orgs like us who use helm can manage dashboards declaratively, and would find this valuable.  We'd like git to be the source of truth, which a new server level setting would facilitate.  Should this issue be reopened, or should a new one be created?

**Comment by jakuboskera** [comment]
Some news regarding this feature?

We have exactly same problem. We would like to allow user to create a dashboard using grafana UI, but disable saving. With this approach user will have to export a JSON model of the dashboard a create CRD GrafanaDashboard in K8s. This will leads that all grafana instance can be declaratively described in git.

**Comment by tanandy** [comment]
We have the same issue, we would like to use only immutable dashboards and not beeing able to save it even for non provisioned

**Comment by holmesb** [comment]
Be good to properly allow gitops workflows in Grafana @torkelo. Is this one on the roadmap? What "more-evidence" do you need?

**Comment by andyspiers** [comment]
This issue also affects us and several times now a developer (who was unaware that they must always check their dashboard JSON into our repo) has spent ages creating a dashboard and because they were just allowed to save it, did not realise that it was only persisted in the local container filesystem only to have it disappear whenever the Grafana pod is rescheduled.

**Comment by odin-of-codin** [comment]
Yes strange this gap hasn't yet been filled. Can implement the perfect gitops workflow, but users can subvert it by just creating a new dashboard and hitting save. If “allowUiUpdates: false” won't achieve, then a new setting is needed. How about “allowPersistNewDashboards: false”? Would default to “true”, so no breaking changes.

**Comment by holmesb** [comment]
Yes strange this gap hasn't yet been filled. Can implement the perfect gitops workflow, but users can subvert it by just creating a new dashboard and hitting save. If “_allowUiUpdates: false_” won't achieve, then a new setting is needed. How about “_allowPersistNewDashboards: false_”? Would default to “true”, so no breaking changes.

**Comment by mdvictor** [comment]
@grafana/grafana-search-and-storage I'm not entirely sure this is dashboards. Could you also have a look over it?

**Comment by Christian-Little-Tava** [comment]
Also interested in this!

**Comment by andyspiers** [comment]
Not sure if commenting now will re-open this issue but in my opinion this is still a required feature
